lab3/README.md

```diff
@@ -0,0 +1,208 @@
+
+Each NF, except for the `UPF`, is assigned an IP address within a `privnet` for internal communication. And assign it an alias for ease of use.
+```yaml
+networks:
+    privnet:
+    aliases:
+        - udr.free5gc.org
+```
+We create three bridge networks named `n1net`, `n2net`, `n3net` and `n6net`. Our goal is respectively assigned network to `AMF`, `SMF`, `UPF` and `UERANSIM`.
+
+```yaml
+n3net:
+  ipam:
+    driver: default
+    config:
+      - subnet: 10.100.3.0/24
+  driver_opts:
+    com.docker.network.bridge.name: br-n3
+
+n4net:
+  ipam:
+    driver: default
+    config:
+      - subnet: 10.100.4.0/24
+  driver_opts:
+    com.docker.network.bridge.name: br-n4
+
+n6net:
+  ipam:
+    driver: default
+    config:
+      - subnet: 10.100.6.0/24
+  driver_opts:
+    com.docker.network.bridge.name: br-n6
+    com.docker.network.container_iface_prefix: dn
+```
+
+Here is the sample of assign `n3net` to upf :
+```yaml
+networks:
+  n3net:
+    aliases:
+      - upf.n3.org
+    ipv4_address: 10.100.3.100
+```
+
+After assigning the networks, it is necessary to update the IP addresses in the corresponding NF configurations.
+
+In `gnbcfg.yaml`, `amfcfg.yaml`, `smfcfg.yaml` and `upfcfg.yaml`, you can find the settings to configure IP addresses as follows.
+```yaml
+  pfcp: # the IP address of N4 interface on this SMF (PFCP)
+    # addr config is deprecated in smf config v1.0.3, please use the following config
+    nodeID: update here # the Node ID of this SMF
+    listenAddr: update here # the IP/FQDN of N4 interface on this SMF (PFCP)
+    externalAddr: update here # the IP/FQDN of N4 interface on this SMF (PFCP)
+```
+Please replace `update here` with the configured N2, N3, and N4 addresses.
+
+Tips: 
+In `smfcfg.yaml`, you will configure the `UPF` N3 interface address because it is required for setting up sessions during SM context creation. If you only use an alias when configuring this address, it may cause DNS resolution issues. Therefore, in `deploy_exercise.yaml`, you should set a static IP address for the `UPF` N3 network and use it here.
+
+After configuring, clone [free5gc-compose](https://github.com/free5gc/free5gc-compose). Then move `free5GLab/lab3/exercise/deploy_exercise.yaml` to `free5gc-compose/` and copy the contents of the files from the `free5GLab/lab3/exercise/config` directory to `free5gc-compose/config`.
+
+You can use these commands to start or stop docker compose.
+```sh
+ce ~/free5gc-compose
+
+// start
+docker compose -f deploy_exercise.yaml up
+
+// remove
+docker compose -f deploy_exercise.yaml down
+```
+
+Refer [Create Subscriber via Webconsole](https://free5gc.org/guide/Webconsole/Create-Subscriber-via-webconsole/#5-add-new-subscriber) to create subscriber 
+
+And, attach to ueransim container and run ue
+```sh
+// attach to container
+docker exec -it ueransim bash
+
+// run ue
+./nr-ue -c config/uecfg.yaml
+```
+And the expected result looks like:
+```sh
```

Comment:
I think you forgot to create a subscriber.
However, you could add links to document in [free5gc.org](https://free5gc.org/)